paasta_tools/cli/utils.py

```diff
@@ -305,7 +305,7 @@ def guess_instance(service, cluster, args):
     else:
         try:
             instances = list_all_instances_for_service(
-                service=service, cluster=cluster, instance_type=None, soa_dir=args.yelpsoa_config_root)
```

Comment:
This isn't the best. The reasoning behind this guess function is to save people the trouble of typing in an instance name. By removing the cluster input the guess is not as good. For example, if you ran something like:

    paasta local-run -c cluster1

and you had an instance called "main" in cluster1 and "backup" in cluster2, this new change would make the guess function suggested "backup" instead of "main" :(